services/scraper/scraper.go

```diff
@@ -0,0 +1,140 @@
+package scraper
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+
+	"github.com/gocolly/colly"
+	"github.com/gocolly/colly/queue"
+)
+
+var selectors = map[string]string {
+	"wholePage": "html",
+	"topImageAds": "#tvcap .plantl a.pla-unit-title-link",
+	"topLinkAds": "#tvcap .d5oMvf > a",
+	"sideImageAds": ".cu-container a.plantl.clickable-card",
+	"nonAds": "#search .yuRUbf > a",
+	"bottomAds": "#bottomads .uEierd > a",
+}
+
+// TODO: Replace this struct with model
+type Result struct {
+	Keyword string
+	TopAdLinks []string
+	OtherAdLinks []string
+	NonAdLinks []string
+	PageCache string
+}
+var results = map[string]*Result{}
+
+const GOOGLE_SEARCH_URL = "http://www.google.com/search?q=%s"
+
+func Search(keywords []string) {
+	collector := colly.NewCollector(colly.Async(true))
+
+	q, err := queue.New(2, &queue.InMemoryQueueStorage{MaxSize: 10000})
```

Comment:
Based on our convention, we should use a long, meaningful variable for local/global/argument variable
```suggestion
	queue, err := queue.New(2, &queue.InMemoryQueueStorage{MaxSize: 10000})
```

![Screen Shot 2021-05-10 at 5 18 40 PM](https://user-images.githubusercontent.com/6076762/117644693-f3a6f680-b1b3-11eb-9e99-e05f89e54be1.png)

Reply:
@junan I don't think we can use `queue` here because it's going to be duplicated with the imported Colly's `queue` so I'll rename it to `searchQueue` 😅

Reply:
fixed on [c143385](https://github.com/carryall/go-google-scraper-challenge/pull/34/commits/c143385da481d2b503193d4ffe7be393c8d5bb12)